src/coreclr/nativeaot/Runtime/windows/PalRedhawkInline.h

```diff
@@ -42,6 +42,13 @@ FORCEINLINE int32_t PalInterlockedExchange(_Inout_ int32_t volatile *pDst, int32
     return _InterlockedExchange((long volatile *)pDst, iValue);
 }
 
+EXTERN_C int64_t _InterlockedExchange64(int64_t volatile *, int64_t);
+#pragma intrinsic(_InterlockedExchange64)
+FORCEINLINE int64_t PalInterlockedExchange64(_Inout_ int64_t volatile *pDst, int64_t iValue)
+{
+    return _InterlockedExchange64(pDst, iValue);
+}
```

Comment:
```suggestion
#ifdef HOST_X86
FORCEINLINE int64_t PalInterlockedExchange64(_Inout_ int64_t volatile *pDst, int64_t iValue)
{
    int64_t iOldValue;

    do {
        iOldValue = *pDst;
    } while (PalInterlockedCompareExchange64(Target,
                                          iValue,
                                          iOldValue) != iOldValue);

    return iOldValue;
}
#else // HOST_X86
EXTERN_C int64_t _InterlockedExchange64(int64_t volatile *, int64_t);
#pragma intrinsic(_InterlockedExchange64)
FORCEINLINE int64_t PalInterlockedExchange64(_Inout_ int64_t volatile *pDst, int64_t iValue)
{
    return _InterlockedExchange64(pDst, iValue);
}
#endif // HOST_X86
```
Try this and move it after `PalInterlockedCompareExchange64` so that it can use `PalInterlockedCompareExchange64`.